uaclient/apt.py

```diff
@@ -626,20 +626,31 @@ def update_esm_caches(cfg) -> None:
 
     import apt  # type: ignore
 
+    from uaclient.actions import status
     from uaclient.entitlements.entitlement_status import ApplicationStatus
     from uaclient.entitlements.esm import (
         ESMAppsEntitlement,
         ESMInfraEntitlement,
     )
 
+    current_status = status(cfg)
+    for service in current_status.get("services", []):
+        if service.get("name", "") == "esm-apps":
+            apps_available = service.get("available", "no")
+        if service.get("name", "") == "esm-infra":
+            infra_available = service.get("available", "no")
```

Comment:
Small optional stylistic suggestion: make these bools here rather than later
```suggestion
            apps_available = service.get("available", "no") == "yes"
        if service.get("name", "") == "esm-infra":
            infra_available = service.get("available", "no") == "yes"
```